### PR TITLE
Fix recently failing "API on WP Latest" job in release testing workflow

### PR DESCRIPTION
**Pull request by rodelgc**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

This PR fixes the broken "API on WP Latest" job in our release tests.

Cause of the failure: Our "Smoke test release" workflow doesn't download the chromium browser, which is needed by the API tests in order to execute the recently added `plugins/woocommerce/tests/api-core-tests/global-setup.js`. So I added a step to download it. But a new problem came after: The global setup consistently fails when logging in to wp-admin. It was difficult to debug because `global-setup.js` doesn't provide traces, screenshots, and videos for debugging.

Final solution I went with: I created two files to be used by the "Smoke test release" workflow:
1. A dedicated playwright config named `ci-release.playwright.config.js`, and
2. A "Setup" Playwright project that's similar to what `global-setup.js` does, but creates traces, screenshots, and videos for debugging. This project uses the `ci-release.global-setup.js` file.
    - This global setup file uses Playwright's `request` object to simplify downloading of the required WooCommerce buid zip without using the `utils` folder from the E2E tests.

I didn't touch the `global-setup.js` file because it's being used in the daily tests and might mess things up there, and also to keep the focus of this PR on fixing the API tests on the release testing workflow. It would be good to have a single `global-setup.js` and Playwright config exclusively for CI runs. I'm planning to submit a PR to do just that after this.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Verify that the check "Run API tests" in this PR passes.
4. Go to [Smoke test release](https://github.com/woocommerce/woocommerce/actions/workflows/smoke-test-release.yml).
5. In the "Run workflow" menu, select this branch.
6. Enter any recent `8.2` release tag like `8.2.0-a.1` into the `WooCommerce Release Tag` field.
7. Click the green "Run workflow" button.
8. Wait for the workflow to finish.
9. Verify that the "API on WP Latest" job passes.
    

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/tests/api-core-tests/ci-release.global-setup.js
Comment (alopezari) [inline plugins/woocommerce/tests/api-core-tests/ci-release.global-setup.js]: This can be done easily with:
```
const [major, minor] = UPDATE_WC.split('.').map(Number);
```
In the example where const `UPDATE_WC = "8.2.0"`, major should be `8` and minor, `2`. I find this approach clearer and more maintainable, because if for whatever reason we also needed to retrieve the 3rd item (0 in this case), we wouldn't need to add a new line to this, just `[major, minor, fix]`.

File: plugins/woocommerce/tests/api-core-tests/ci-release.global-setup.js
Comment (alopezari) [inline plugins/woocommerce/tests/api-core-tests/ci-release.global-setup.js]: Wouldn't it work here just to handle the `page.waitForResponse` with await instead of doing that in line 68? Is this because you need to upload the plugin, then wait until it has finished uploading and click install?

**Review by alopezari — APPROVED** [review]
Thanks for working on this @rodelgc!

I added a question and a code suggestion (not mandatory). However, the test indicated in the testing instructions [passed](https://github.com/woocommerce/woocommerce/actions/runs/6402049580):
image

So I'll approve this PR but won't merge it so that you can decide if you want to implement the suggestion or just merge it instead.

**Comment by nigeljamesstevenson** [comment]
Hi @rodelgc - based on the comment from @alopezari https://github.com/woocommerce/woocommerce/pull/40453#pullrequestreview-1656695442, I'll let you decide how you want to proceed 😊

File: plugins/woocommerce/tests/api-core-tests/ci-release.global-setup.js
Comment (rodelgc) [reply plugins/woocommerce/tests/api-core-tests/ci-release.global-setup.js]: Yeah this is a great suggestion! Thanks for spotting this. Addressed in bc6409b5d5f77b939b90b23c7fff2739852abce5

File: plugins/woocommerce/tests/api-core-tests/ci-release.global-setup.js
Comment (rodelgc) [reply plugins/woocommerce/tests/api-core-tests/ci-release.global-setup.js]: > Wouldn't it work here just to handle the page.waitForResponse with await instead of doing that in line 68?

I think it would still work. The reason I separated line 64 and line 68 was because I was just following the format [here](https://playwright.dev/docs/api/class-page#page-wait-for-response). 

I had just pushed a commit that combines those two lines into a single `await page.waitForResponse()` call -- 6e66f48af89b48ab5dad018370821d6c8c7d6a05. Thank you for suggesting this Alex! The code looked a bit more concise with your suggestion.

**Comment by rodelgc** [comment]
> Hi @rodelgc - based on the comment from @alopezari https://github.com/woocommerce/woocommerce/pull/40453#pullrequestreview-1656695442, I'll let you decide how you want to proceed 😊

Thanks @nigeljamesstevenson. I followed Alex's suggestions so I pushed a few more minor commits. Once the checks passed, I'll go ahead and merge this. Thanks so much for taking a look!

**Pull request merged**
